### PR TITLE
Update coverlet, codecov, and ReportGenerator

### PR DESCRIPTION
**Pull request by sharwell**
Enable coverage for all source folders in the repository.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10638)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.cs [hunk 1]
```diff
@@ -14,6 +14,11 @@ namespace System.Windows.Forms.Tests;
 
 public partial class ToolStripTests
 {
+    public ToolStripTests()
+    {
+        typeof(ToolStrip).TestAccessor().Dynamic.LastKeyData = Keys.None;
+    }
```
Comment (sharwell) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.cs:20]: 📝 As best I can tell, the change from xunit runner to VSTest runner changed the test order slightly, and one of the tests that sets LastKeyData to Keys.Left is now running before a `ToolStrip_GetNextItem_CyclesBackwardExpected` test that reads LastKeyData, causing a failure.

File: Directory.Build.props [hunk 2]
```diff
@@ -70,11 +70,10 @@
     <SingleHit>true</SingleHit>
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
     <CoverletOutputFormat>opencover</CoverletOutputFormat>
-    <Include>[System.Windows.*]*,[System.Drawing.Common.*]*,[System.Private.*]*,[Microsoft.VisualBasic.*]*</Include>
-    <!-- Exclude dependencies that embed their symbol information -->
-    <Exclude>[Microsoft.DotNet.XUnitExtensions]*,[InternalUtilitiesForTests*]*,[xunit*]*</Exclude>
-    <!-- Exclude anything tagged with Obsolete or with ExcludeFromCodeCoverage !!Avoid using this!! -->
-    <ExcludeByAttribute>Obsolete,ExcludeFromCodeCoverage</ExcludeByAttribute>
+    <Include></Include>
```
Comment (lonitra) [inline Directory.Build.props:73]: Could you help me understand why this was removed? Does having nothing in both the `include`/`exclude` just automatically includes everything?

Comment (sharwell) [reply Directory.Build.props:73]: > Does having nothing in both the include/exclude just automatically includes everything?

Correct, everything is now included except for code marked with ExcludeFromCodeCoverage.

**Review by lonitra — APPROVED** [review]
Thank you 👍

**Pull request merged**